gax/src/test/java/com/google/api/gax/batching/BatcherImplTest.java

```diff
@@ -711,6 +713,123 @@ public void run() {
     }
   }
 
+  @Test
+  public void testConstructors() throws InterruptedException {
+    try (BatcherImpl batcher1 =
+        new BatcherImpl<>(
+            SQUARER_BATCHING_DESC_V2,
+            callLabeledIntSquarer,
+            labeledIntList,
+            batchingSettings,
+            EXECUTOR)) {
+      assertThat(batcher1.getFlowController()).isNotNull();
+      assertThat(batcher1.getFlowController().getLimitExceededBehavior())
+          .isEqualTo(batchingSettings.getFlowControlSettings().getLimitExceededBehavior());
+      assertThat(batcher1.getFlowController().getMaxOutstandingElementCount())
+          .isEqualTo(batchingSettings.getFlowControlSettings().getMaxOutstandingElementCount());
+      assertThat(batcher1.getFlowController().getMaxOutstandingRequestBytes())
+          .isEqualTo(batchingSettings.getFlowControlSettings().getMaxOutstandingRequestBytes());
+    }
+
+    FlowController flowController =
+        new FlowController(
+            FlowControlSettings.newBuilder()
+                .setLimitExceededBehavior(LimitExceededBehavior.ThrowException.ThrowException)
+                .setMaxOutstandingRequestBytes(6000L)
+                .build());
+    try (BatcherImpl batcher2 =
+        new BatcherImpl<>(
+            SQUARER_BATCHING_DESC_V2,
+            callLabeledIntSquarer,
+            labeledIntList,
+            batchingSettings,
+            EXECUTOR,
+            flowController)) {
+      assertThat(batcher2.getFlowController()).isSameInstanceAs(flowController);
+    }
+  }
+
+  @Test
+  public void testThrottlingBlocking() throws Exception {
+    BatchingSettings settings =
+        BatchingSettings.newBuilder()
+            .setElementCountThreshold(1L)
+            .setRequestByteThreshold(1L)
+            .build();
+    FlowController flowController =
+        new FlowController(
+            FlowControlSettings.newBuilder()
+                .setLimitExceededBehavior(LimitExceededBehavior.Block)
+                .setMaxOutstandingElementCount(1L)
+                .build());
+    ExecutorService executor = Executors.newSingleThreadExecutor();
+    try (final Batcher<Integer, Integer> batcher =
+        new BatcherImpl<>(
+            SQUARER_BATCHING_DESC_V2,
+            callLabeledIntSquarer,
+            labeledIntList,
+            settings,
+            EXECUTOR,
+            flowController)) {
+      flowController.reserve(1, 1);
+      Future future =
+          executor.submit(
+              new Runnable() {
+                @Override
+                public void run() {
+                  batcher.add(1);
+                }
+              });
+      try {
+        future.get(100, TimeUnit.MILLISECONDS);
```

Comment:
If `TimeoutException` is expected, can we make this shorter (i.e. since we are guaranteed to wait, lets wait as little as possible, otherwise it may quickly lead to really long-running unit tests, which should not happen). This comments/question applies to all timed `get()`s in this test.